web-ng/src/app/components/map/map.component.spec.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import firebase from 'firebase/app';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { MapComponent } from './map.component';
+import { MatDialog } from '@angular/material/dialog';
+import { ProjectService } from '../../services/project/project.service';
+import { FeatureService } from '../../services/feature/feature.service';
+import { NavigationService } from '../../services/navigation/navigation.service';
+import { Project } from '../../shared/models/project.model';
+import { Feature, LocationFeature } from '../../shared/models/feature.model';
+import { StringMap } from '../../shared/models/string-map.model';
+import { Map, List } from 'immutable';
+import { Layer } from '../../shared/models/layer.model';
+import { of } from 'rxjs';
+import { GoogleMapsModule } from '@angular/google-maps';
+
+// import { initialize } from "@googlemaps/jest-mocks";
+// import createGoogleMapsMock from 'jest-google-maps-mock';
+
+describe('MapComponent', () => {
+  let component: MapComponent;
+  let fixture: ComponentFixture<MapComponent>;
+  const dialog: Partial<MatDialog> = {};
+  const map: jasmine.SpyObj<google.maps.Map> = jasmine.createSpyObj(
```

Comment:
IIuc, this uses as real Map instance but stubs out selected methods. This still makes the test dependent on the impl of Map. Can we use a fake here instead, declaring stubs only for the methods we use?